src/ekomark/benchmark/runner.py

```diff
@@ -245,10 +246,11 @@ def log(self, theory, _, pdf, me, ext):
                 flavor_rotation=rotate_to_evolution,
                 qed=qed,
             )
-        for q2 in q2s:
+        for (q2, ref_pdfs), (q2a, res) in zip(ext["values"].items(), pdf_grid.items()):
+            np.testing.assert_allclose(q2, q2a)
```

Comment:
```suggestion
        np.assert_allclose(ext["values"].keys(), pdf_grid.keys())
        meaningful_name = zip(ext["values"].items(), pdf_grid.values())
        for (q2, ref_pdfs), res in meaningful_name:
```
To marginally simplify, you could split the check on its own (assuming NumPy is supporting your aggregated `.keys()` with allclose).

Usage of meaningful names always helps, making it clear at first site.

All in all, it is pretty fine also as it as.

Reply:
fine ... but then I prefer my solution